titus_isolate/allocate/forecast_ip_cpu_allocator.py

```diff
@@ -68,41 +70,73 @@ def __init__(self,
         self.__config_manager = config_manager
         self.__cnt_rebalance_calls = 0
         self.__event_log_manager = None
-
+        self.__call_meta = None # track things __place_threads call
+
+
+    def __with_report(func):
+        @wraps(func)
+        def wrapped(inst, cpu: Cpu, workload_id: str, workloads: dict, cpu_usage: dict, instance_id: str):
+            inst.__call_meta = {}
+            report = lambda cpu_ : inst.report_cpu_event(
+                inst.__event_log_manager,
+                cpu_,
+                list(workloads.values()),
+                cpu_usage,
+                instance_id,
+                inst.__call_meta) 
+            try:
+                cpu = func(inst, cpu, workload_id, workloads, cpu_usage, instance_id)
+                report(cpu)
+                return cpu
+            except Exception as e:
+                report(cpu)
```

Comment:
How do you distinguish in the event between success/failure in generated events?

Reply:
here: https://github.com/Netflix-Skunkworks/titus-isolate/pull/154/files#diff-46a2ba31c28d44e3d925a297228d956dR363